channeldb/payments_test.go

```diff
@@ -213,9 +254,29 @@ func TestRouteSerialization(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// First we verify all the records match up porperly, as they aren't
+	// able to be properly compared using reflect.DeepEqual.
+	for i := 0; i < len(testRoute.Hops); i++ {
+		for j := 0; j < len(testRoute.Hops[i].TLVRecords); j++ {
+			expectedRecord := testRoute.Hops[i].TLVRecords[j]
+			newRecord := route2.Hops[i].TLVRecords[j]
+
+			err := assertHopRecordsEqual(expectedRecord, newRecord)
+			if err != nil {
+				t.Fatalf("records not equal: %v", err)
+			}
+		}
+	}
+
+	// Now that we know the records match up, we'll examine the remainder
+	// of the route without the TLV records attached.
+	testRoute.Hops[0].TLVRecords = nil
+	testRoute.Hops[1].TLVRecords = nil
+	route2.Hops[0].TLVRecords = nil
+	route2.Hops[1].TLVRecords = nil
+
```

Comment:
Ok, something is wrong with `DeepEqual` on the records? Yes, I think one function that does an all-inclusive compare of routes would be nice in this test.